src/map/lua/lua_baseentity.cpp

```diff
@@ -831,20 +831,38 @@ void CLuaBaseEntity::startEvent(uint32 EventID, sol::variadic_args va)
         }
     }
 
-    uint32 param0 = va.get_type(0) == sol::type::number ? va.get<uint32>(0) : 0;
-    uint32 param1 = va.get_type(1) == sol::type::number ? va.get<uint32>(1) : 0;
-    uint32 param2 = va.get_type(2) == sol::type::number ? va.get<uint32>(2) : 0;
-    uint32 param3 = va.get_type(3) == sol::type::number ? va.get<uint32>(3) : 0;
-    uint32 param4 = va.get_type(4) == sol::type::number ? va.get<uint32>(4) : 0;
-    uint32 param5 = va.get_type(5) == sol::type::number ? va.get<uint32>(5) : 0;
-    uint32 param6 = va.get_type(6) == sol::type::number ? va.get<uint32>(6) : 0;
-    uint32 param7 = va.get_type(7) == sol::type::number ? va.get<uint32>(7) : 0;
+    std::vector<std::pair<uint8, uint32>> params;
 
-    int16 textTable = va.get_type(8) == sol::type::number ? va.get<int16>(8) : -1;
+    int16 textTable = -1;
+    if (va.get_type(0) == sol::type::table)
+    {
+        auto table = va.get<sol::table>(0);
+        for (int i = 0; i < 8; i++)
+        {
+            uint32 param = table.get_or<uint32>(i, 0);
+            if (param != 0)
+            {
+                params.emplace_back(i, param);
+            }
+        }
+
+        textTable = table.get_or<int16>("text_table", -1);
+    }
+    else
+    {
+        for (int i = 0; i < 8; i++)
```

Comment:
This makes so much more sense than what we were doing before... It was a long conversion...